notebooks/tests/test_example.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+import yaml
+from testbook import testbook
+
+
+# Run the YAML config section of the notebook.
+@testbook("./example.ipynb", execute=range(0, 4))
+def test_get_yaml_config(tb):
+
+    # Check output text of the cell that prints the YAMLconfig object.
+    assert (
+        tb.cell_output_text(3)
+        == "values:\n  date: 20240105\n  greeting: Good Night\n  recipient: Moon\n  repeat: 2"
+    )
+
+    # Extract the config_yaml variable from the notebook and test its values.
+    nb_yaml = tb.ref("config_yaml")
+    assert nb_yaml["values"] == {
+        "date": 20240105,
+        "greeting": "Good Night",
+        "recipient": "Moon",
+        "repeat": 2,
+    }
+
+
+def test_template_render():
+    # Remove the rendered file if it exists.
+    rendered_path = "./fixtures/rendered_config.yaml"
+    if os.path.exists(rendered_path):
+        os.remove(rendered_path)
+
+    # Run the template rendering section of the notebook.
+    with testbook("./example.ipynb", execute=range(4, 14)) as tb:
```

Comment:
I'm a little wary of supporting these numbers (in this line and below) as we add/remove examples or even change the order of the boxes. 

Are there ways that we might be able to label the different cells and refer to the labels instead of relying on the order of the cells?

Reply:
There isn't a way to target a label of the cells as far as I can see in the [docs,](https://testbook.readthedocs.io/en/latest/usage/index.html#using-execute-to-control-which-cells-are-executed-before-test) but using `execute=True` is an option that would execute the entire notebook without the need to specify any cells. This would probably work for most cases, especially if the notebooks are relatively small.

Reply:
This might be a good reason to create many small notebooks that each focus on one small portion of the API (config, file, template, ect.).

Reply:
I updated these calls to execute all cells in the notebook.